core/src/nucleus/ribosome/api/link_entries.rs

```diff
@@ -1,31 +1,30 @@
 use dht::actions::add_link::*;
 use futures::executor::block_on;
-use holochain_core_types::error::HolochainError;
 use holochain_wasm_utils::api_serialization::link_entries::LinkEntriesArgs;
 use nucleus::ribosome::{api::ZomeApiResult, Runtime};
-use serde_json;
+use std::convert::TryFrom;
 use wasmi::{RuntimeArgs, RuntimeValue};
 
 /// ZomeApiFunction::LinkEntries function code
 /// args: [0] encoded MemoryAllocation as u32
 /// Expected complex argument: LinkEntriesArgs
-/// Returns 'Success' return code on success.
 pub fn invoke_link_entries(runtime: &mut Runtime, args: &RuntimeArgs) -> ZomeApiResult {
     // deserialize args
-    let args_str = runtime.load_utf8_from_args(&args);
-    let input: LinkEntriesArgs = match serde_json::from_str(&args_str) {
+    let args_str = runtime.load_json_string_from_args(&args);
+    let input = match LinkEntriesArgs::try_from(args_str.clone()) {
         Ok(entry_input) => entry_input,
         // Exit on error
-        Err(_) => return ribosome_error_code!(ArgumentDeserializationFailed),
+        Err(_) => {
+            println!(
+                "invoke_link_entries failed to deserialize LinkEntriesArgs: {:?}",
+                args_str
+            );
+            return ribosome_error_code!(ArgumentDeserializationFailed);
+        }
```

Comment:
this is intentional because there is no other way to know what went wrong

the error code by itself doesn't contain enough information to actually debug anything

debugging this downstream involves rebuilding two or three repos per change

i agree that the println! is not ideal, but there does need to be a way to get debug information out, so let's discuss in a followup